workflow/common/util.go

```diff
@@ -241,26 +241,50 @@ func GetExecutorOutput(exec remotecommand.Executor) (*bytes.Buffer, *bytes.Buffe
 // * local parameters (e.g. {{pod.name}})
 func ProcessArgs(tmpl *wfv1.Template, args wfv1.Arguments, globalParams, localParams map[string]string, validateOnly bool) (*wfv1.Template, error) {
 	// For each input parameter:
-	// 1) check if was supplied as argument. if so use the supplied value from arg
-	// 2) if not, use default value.
-	// 3) if no default value, it is an error
+	// 1) check if it's a passthrough regexp, in which case substitute with matching arguments
+	// 2) check if was supplied as argument. if so use the supplied value from arg
+	// 3) if not, use default value.
+	// 4) if no default value, it is an error
 	tmpl = tmpl.DeepCopy()
-	for i, inParam := range tmpl.Inputs.Parameters {
-		if inParam.Default != nil {
-			// first set to default value
-			inParam.Value = inParam.Default
-		}
-		// overwrite value from argument (if supplied)
-		argParam := args.GetParameterByName(inParam.Name)
-		if argParam != nil && argParam.Value != nil {
-			newValue := *argParam.Value
-			inParam.Value = &newValue
-		}
-		if inParam.Value == nil {
-			return nil, errors.Errorf(errors.CodeBadRequest, "inputs.parameters.%s was not supplied", inParam.Name)
+	newInputParameters := make([]wfv1.Parameter, 0)
+
+	for _, inParam := range tmpl.Inputs.Parameters {
+		if inParam.Regexp != "" {
+			argParams, err := args.GetParametersByRegexp(inParam.Regexp)
+			if err != nil {
+				return nil, errors.Errorf(errors.CodeBadRequest, "regexp %s could not be processed", inParam.Regexp)
+			}
+			for _, argParam := range argParams {
+				newName := argParam.Name
+				newValue := argParam.Value
+				newRegexp := argParam.PassthroughRegexp //used for workflow templates where all arguments may not have been resolved already
+				newParameter := wfv1.Parameter{Name: newName, Value: newValue, Regexp: newRegexp}
```

Comment:
I think `Regexp: newRegexp` is not necessary as https://github.com/argoproj/argo/pull/1490/files#r308056723.